plugins/tiddlywiki/multiwikiserver/modules/store/sql-tiddler-store.js

```diff
@@ -143,20 +147,25 @@ SqlTiddlerStore.prototype.processOutgoingTiddler = function(tiddlerFields,tiddle
 
 /*
 */
-SqlTiddlerStore.prototype.processIncomingTiddler = function(tiddlerFields) {
+SqlTiddlerStore.prototype.processIncomingTiddler = function(tiddlerFields, metadata) {
 	let attachmentSizeLimit = $tw.utils.parseNumber(this.adminWiki.getTiddlerText("$:/config/MultiWikiServer/AttachmentSizeLimit"));
 	if(attachmentSizeLimit < 100 * 1024) {
 		attachmentSizeLimit = 100 * 1024;
 	}
+	var tiddlerInfo = this.sqlTiddlerDatabase[this.fetchTiddlerFuncMap[metadata.entityType]](metadata.title,metadata.name);	
```

Comment:
I am not keen on the indirection here. I understand the motivation of avoiding code duplication, but the result is almost as complexity as the duplication.

In any case, I think calling getRecipeTiddler or getBagTiddler is overkill here. We only want to know the attachment status of the tiddler, while those methods perform a second query to retrieve the tiddler fields which is not needed in this context.

Therefore I think it would be better to handle things as follows:

* Introduce new database methods getBagTiddlerAttachmentBlob and getRecipeTiddlerAttachmentBlob that just do the lookup of the attachment_blob property
* change processIncomingTiddler to take an additional parameter "attachment_blob" instead of "metadata"
* In the two calling sites of "SqlTiddlerStore.prototype.saveBagTiddler" and "SqlTiddlerStore.prototype.saveBagTiddler", invoke the appropriate one of the new methods to pass the attachment_blob